tests/test_catalog.py

```diff
@@ -837,6 +837,23 @@ def check_all_absolute(cat: Catalog) -> None:
                 c2.catalog_type = CatalogType.ABSOLUTE_PUBLISHED
                 check_all_absolute(c2)
 
+    def test_self_contained_catalog_collection_item_links(self) -> None:
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            cat = TestCases.test_case_9()
+            cat.normalize_hrefs(tmp_dir)
+            cat.validate_all()
+            print(f'Saving catalog to: {tmp_dir}')
+            cat.save(catalog_type=CatalogType.SELF_CONTAINED)
+
+            # read the item back, to make experiment clean
+            item = Item.from_file(f'{tmp_dir}/collection-issue-657/item-issue-657/item-issue-657.json')
+            # ensure that all links in the output json are relative
+            # everything of a nested level > 2 is not relative?
+            for link in item.links:
+                self.assertFalse(is_absolute_href(link.target))
```

Comment:
Is it even a valid test? pystac can fix links somehow... to make this test even more explicit I can load item as a json dict 🦐

Reply:
Yeah, `link.target` can be either a string or a `STACObject` instance, but `is_absolute_href` expects only strings, so this is raising a `mypy` error. It might be best to do something like:

```python
for link in item.links:
    href = link.get_href()
    self.assertIsNotNone(href)
    self.assertFalse(is_absolute_href(href))
```

Reply:
done! thanks